lib/vsc/install/shared_setup.py

```diff
@@ -174,6 +175,161 @@ def run(self):
         orig_bdist_rpm.run(self)
 
 
+def filter_testsuites(testsuites):
+    """(Recursive) filtering of (suites of) tests"""
+    test_filter = getattr(__builtin__, '__test_filter')['function']
+
+    res = type(testsuites)()
+
+    for ts in testsuites:
+        # ts is either a test or testsuite of more tests
+        if isinstance(ts, TestSuite):
+            res.addTest(filter_testsuites(ts))
+        else:
+            if re.search(test_filter, ts._testMethodName):
+                res.addTest(ts)
+    return res
+
+
+class VscScanningLoader(ScanningLoader):
+    """The class to look for tests"""
+
+    TEST_LOADER_MODULE = __name__
+
+    def loadTestsFromModule(self, module):
+        """
+        Support test module and function name based filtering
+        """
+        testsuites = ScanningLoader.loadTestsFromModule(self, module)
+
+        test_filter = getattr(__builtin__,'__test_filter')
+
+        res = testsuites
+
+        if test_filter['module'] is not None:
+            name = module.__name__
+            if name in test_filter['allowmods']:
+                # a parent name space
+                pass
+            elif re.search(test_filter['module'], name):
+                if test_filter['function'] is not None:
+                    res = filter_testsuites(testsuites)
+                # add parents (and module itself)
+                pms = name.split('.')
+                for pm_idx in range(len(pms)):
+                    pm = '.'.join(pms[:pm_idx])
+                    if not pm in test_filter['allowmods']:
+                        test_filter['allowmods'].append(pm)
+            else:
+                res = type(testsuites)()
+
+        return res
+
+
+class VscTestCommand(TestCommand):
+    """
+    The cmdclass for testing
+    """
+
+    # make 2 new 'python setup.py test' options available:
+    #    --test-filterf and --test-filertm
```

Comment:
either you mention both long/short options here, or you don't mention them at all